component/prometheus/remotewrite/remote_write.go

```diff
@@ -107,6 +107,9 @@ func New(o component.Options, c Arguments) (*Component, error) {
 			if localID == 0 {
 				prometheus.GlobalRefMapping.GetOrAddLink(res.opts.ID, uint64(newRef), l)
 			}
+			//TODO(thampiotr): We can expose this as `notify_on_append` option - it can be used to reduce latency in
+			// the pipeline if users want to, with a trade-off being more CPU and I/O.
+			remoteStore.Notify()
```

Comment:
Adding this shortens the time it takes to run prometheus tests from 15s to 5s. But I don't see a reason why these tests wouldn't run in 200ms each, we just may need to do some digging.